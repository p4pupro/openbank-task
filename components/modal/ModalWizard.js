import React, { useState } from "react";
import { useForm, useStep } from "react-hooks-helper";
import { Info } from "../info/Info";
import { CreatePassword } from "../create/CreatePassword";
import { Feedback } from "../feedback/Feedback";

const steps = [
  { id: "productInfo" },
  { id: "createPassword" },
  { id: "feedback" },
];

const defaultData = {
  password: "",
  confirmPassword: "",
  rememberPassword: "",
};
export const ModalWizard = ({ onRequestClose }) => {
  const [formData, setForm] = useForm(defaultData);
  const { step, navigation } = useStep({ initialStep: 0, steps });
  const { id } = step;
  const [response, setResponse] = useState(401);

  const props = {
    formData,
    setForm,
    navigation,
    onRequestClose,
    response,
    setResponse,
  };

  switch (id) {
    case "productInfo":
      return <Info {...props} />;
    case "createPassword":
      return <CreatePassword {...props} />;
    case "feedback":
      return <Feedback {...props} />;
    default:
      return null;
  }
};
