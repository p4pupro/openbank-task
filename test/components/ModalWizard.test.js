import React from "react";
import "@testing-library/jest-dom";
import { shallow, ShallowWrapper } from "enzyme";
import { ModalWizard } from "../../components/modal/ModalWizard";

describe("Test - <ModalWizard />", () => {
  const toggleModal = jest.fn();
  let wrapper = shallow(<ModalWizard onRequestClose={toggleModal} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<ModalWizard onRequestClose={toggleModal} />);
  });

  test("Should renders fine", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
