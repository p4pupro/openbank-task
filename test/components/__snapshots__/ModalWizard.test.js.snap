// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test - <ModalWizard /> Should renders fine 1`] = `
<Info
  formData={
    Object {
      "confirmPassword": "",
      "password": "",
      "rememberPassword": "",
    }
  }
  navigation={
    Object {
      "go": [Function],
      "next": [Function],
      "pause": [Function],
      "play": [Function],
      "previous": [Function],
    }
  }
  onRequestClose={[MockFunction]}
  response={401}
  setForm={[Function]}
  setResponse={[Function]}
/>
`;
